### PR TITLE
dns: add dns.rcode keyword - v8

### PR DESCRIPTION
**Pull request by hadiqaalamdar**
Feature #6621

- [x] I have read the contributing guide lines at https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/6621) ticket: https://redmine.openinfosecfoundation.org/issues/6621

Previous PR: https://github.com/OISF/suricata/pull/10249

Describe changes:
- made the suggested changes from the previous PR
- made changes to documentation
- successfully ran the following tests:
- from rust dir: make check
- from rust dir: cargo test
- from rust dir: cargo clippy --all-features
- from rust dir: rustfmt src/dns/*
- from suricata dir: ./scripts/clang-format.sh check-branch
- from suricata dir: ./scripts/clang-format.sh branch
- from suricata dir running ./src/suricata -u showed that one unit test had failed. 
- from suricata dir running ./src/suricata -u -U dns showed 7 tests passed.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1578

### CODE REVIEW

**Comment by jasonish** [comment]
Rebasing your S-V branch should clean up the CI errors.

File: doc/userguide/rules/dns-keywords.rst [hunk 1]
```diff
@@ -57,6 +57,36 @@ Match on DNS requests where the **opcode** is not between 7 and 15:
...
+dns.rcode
+---------
+
+This keyword matches on the **rcode** found in the DNS header flags.
+
+dns.opcode uses an :ref:`unsigned 8-bit integer <rules-integer-keywords>`.
+
+Currently suricata only supports rcode values in the range [0-15] while the current DNS version supports rcode values from [0-23] as specified in RFC6895. 
+We plan to extend the the rcode values supported by suricata according to RFC6895 as tracked by the ticket: https://redmine.openinfosecfoundation.org/issues/6650
+Link to RFC6895 documentation: https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-6
```
[the remark this answers is not in the document]
Comment (jufajardini) [reply doc/userguide/rules/dns-keywords.rst:69]: This also probably isn't rendering as expected. If you want the docs to have line breaks, a blank line is needed in between the paragraphs.

[the remark this answers is not in the document]
Comment (jufajardini) [reply doc/userguide/rules/dns-keywords.rst:69]: Some suggestions:
```suggestion
Currently, Suricata only supports rcode values in the range [0-15], while the current DNS version supports rcode values from [0-23] as specified in `RFC 6895<https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-6>`_.

We plan to extend the rcode values supported by Suricata, this is tracked by the ticket: https://redmine.openinfosecfoundation.org/issues/6650.
```

**Review by jufajardini — CHANGES_REQUESTED** [review]
Code changes seem good to me, but the docs can be improved a bit :)

We'll need to rebase the SV PR (and submit a new one) for CI to get green again, too.

**Comment by hadiqaalamdar** [comment]
New PR: https://github.com/OISF/suricata/pull/10458

**Pull request closed without merging**